src/main/java/org/wildfly/security/mechanism/MechanismUtil.java

```diff
@@ -64,7 +65,7 @@ private MechanismUtil() {}
      * @param <S> the password type
      * @return the password
      */
-    public static <S extends Password> S getPasswordCredential(String userName, CallbackHandler callbackHandler, Class<S> passwordType, String passwordAlgorithm, AlgorithmParameterSpec matchParameters, AlgorithmParameterSpec generateParameters, Supplier<Provider[]> providers) throws AuthenticationMechanismException {
+    public static <S extends Password> S getPasswordCredential(String userName, CallbackHandler callbackHandler, Class<S> passwordType, String passwordAlgorithm, AlgorithmParameterSpec matchParameters, AlgorithmParameterSpec generateParameters, Supplier<Provider[]> providers, ElytronMessages log) throws AuthenticationMechanismException {
```

Comment:
Unfortunately this class is also public API as well so we can't just change the message signature.